lib_xua/src/core/endpoint0/xua_endpoint0.c

```diff
@@ -951,13 +972,83 @@ void XUA_Endpoint0_loop(XUD_Result_t result, USB_SetupPacket_t sp, chanend c_ep0
     } /* if(result == XUD_RES_OKAY) */
 
     {
+#if XUA_ENABLE_BOS_DESC
+        if(result == XUD_RES_ERR)
+        {
+            unsigned bmRequestType = (sp.bmRequestType.Direction<<7) | (sp.bmRequestType.Type<<5) | (sp.bmRequestType.Recipient);
+            if(bmRequestType == USB_BMREQ_D2H_VENDOR_DEV)
+            {
+                if((sp.bRequest == REQUEST_GET_MS_DESCRIPTOR) &&
+                    sp.wIndex == MS_OS_20_DESCRIPTOR_INDEX)
+                {
+                    if(DFU_mode_active)
+                    {
+                        result = XUD_DoGetRequest(ep0_out, ep0_in, (unsigned char*)desc_ms_os_20_dfu, MS_OS_20_DESC_LEN_DFU, sp.wLength);
+                    }
+                    else
+                    {
+                        result = XUD_DoGetRequest(ep0_out, ep0_in, (unsigned char*)desc_ms_os_20_runtime, MS_OS_20_DESC_LEN_RUNTIME, sp.wLength);
+                    }
+                }
+            }
+
+        }
+#endif
+
+#if XUA_DFU_EN
+        if(result == XUD_RES_ERR)
+        {
+            // In the new Theycon driver implementation, the XMOS_DFU_REVERTFACTORY request comes as a H2D vendor request and not as a class request addressed to the DFU interface
```

Comment:
Would be nice to replace "new" with "since driver version X"

Reply:
done